### ISSUE TITLE
bug(photomath): Patching "unlock-plus" fails on apk version 8.21.0

### ISSUE THREAD

**Issue by Edgiest05** [revanced/revanced-patches#1852]
### Type

Error while patching

### Bug description

Attempting to patch latest Photomath play store version (8.21.0) fails

### Steps to reproduce

Step 1. Select Photomath application in Revanced Manager
Step 2. Choose "unlock-plus" patch
Step 3. Apply patch (fails in progress)

### Relevant log output

```shell
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
Deleting existing resource cache directory
Decoding resources
Applied predictive-back-gesture
Applied remove-screenshot-restriction
Failed to apply unlock-plus: 'unlock-plus' depends on 'SignatureDetectionPatch' but the following error was raised: java.lang.NullPointerException
	at app.revanced.patches.photomath.detection.signature.patch.SignatureDetectionPatch.execute(SignatureDetectionPatch.kt:30)
	at app.revanced.patches.photomath.detection.signature.patch.SignatureDetectionPatch.execute(SignatureDetectionPatch.kt:17)
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:328)
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:301)
	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:36)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:350)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:127)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-20(MainActivity.kt:321)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$kxvfcFiIKARIH6oxYjk9gWWxjp0(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:22)
	at java.lang.Thread.run(Thread.java:1012)
Repacking patched apk
Compiling resources
Writing modified dex files
Finished!
```

### Screenshots or videos

![Choosing Patch](https://user-images.githubusercontent.com/90446383/229915176-2223a569-8775-4140-bad9-3d0ca1c6b5c7.jpg)
![Output_1](https://user-images.githubusercontent.com/90446383/229915204-07f3b7e5-ec77-46db-ab5f-fa82c1282d64.jpg)
![Output_2](https://user-images.githubusercontent.com/90446383/229915221-526954f8-d554-4c2c-a934-d37a07c7df2f.jpg)

### Additional context

Android version: 12
Revanced manager version: 0.0.57
Photomath version: 8.21.0
No custom tweaks to the patching process

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by EhabxAlaa** [comment]
There is a problem with patching Version 8.21.0 ,but patching 8.20.0 works just fine , just remember to download the apk version and select it from your storage( don't install it and then try to patch )

**Comment by Edgiest05** [comment]
The workaround is much appreciated (thanks EhabxAlaa). I'll keep the issue open but with a more precise title.

**Comment by badawoll** [comment]
The fix has been implemented in #1886, now awaiting merge to `main`.

**Comment by Edgiest05** [comment]
Amazing, I'll close the issue. Thanks badawoll for telling me.
